src/handler.rs

```diff
@@ -182,69 +179,64 @@ struct ObsBuildInfo {
 }
 
 #[derive(Clone, Debug)]
-pub struct StringFile {
+pub struct ObsArtifact {
```

Comment:
Wait, now `StringFile` is gone and there’s `ObsArtifact` instead… I’m a bit confused.

Reply:
The first commit was to move away from passing around open temp files. The second was moving towards storing artifacts in memory.
I was wondering whether it's necessary to interact with the file system at all. None of the files are kept around for long.
I assume in the case of logs and build-info.yml data keeping things in memory shouldn't be a problem. I'm not as sure about the binaries, what's their maximum plausible size?
I guess I should squash dab2198f4da2e07d7c620666423fedbeaad0bed6

Reply:
Keeping binaries in memory is not a good idea, they can be really really large.

For instance `gcc-mingw-w64` produces 3.5GB of `.deb` files for each architecture.

The `texlive-extra` package gets to 4.3GB but only because when counting "binaries" OBS also includes the source tarballs, which in this case weights 2.2GB alone.

Reply:
I've squashed and reordered a few commits.